src/protocols/ab/eip_cip.c

```diff
@@ -488,6 +488,13 @@ int build_read_request_connected(ab_tag_p tag, int byte_offset)
 
     req->allow_packing = tag->allow_packing;
 
+    /* set the response size to the size of data from the previous read */
+    req->response_capacity = tag->size;
```

Comment:
Actually... this won't work.  Some tags have more data than fits in a request.   In that case, the tag size is greater than the request buffer size.

Reply:
Yes, this wont work to fetch a single array or UDT which is larger than the response buffer size, it wasn't intended to do that. This PR was intended only to allow packing of many requests for which the response is less than the buffer size.

If the user requests a UDT/array which is too big for the buffer, they already get an easily understood "data too large" type error which I see as being fine. If  however they  enable packing and request many items which are smaller than the buffer size, they will still get a "too large" type error. This seems more like a bug to me and is what this PR is hoping to stop from happening.

There is a potential problem here if the tag->size says that the data will fit and then when it is actually read, the fetched data is larger than this for some reason.